scripts/linters/pylint_extensions.py

```diff
@@ -1922,6 +1932,95 @@ def process_tokens(self, tokens):
                         'single-line-pragma', line=line_num)
 
 
+class TypeIgnoreCommentChecker(checkers.BaseChecker):
+    """Custom pylint checker which checks if MyPy's type ignores are properly
+    documented or not.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    EXCLUDED_DIRS_HAVING_IGNORE_TYPE_COMMENTS = copy.deepcopy(
+        EXCLUDED_TYPE_COMMENT_DIRECTORIES
+    )
+
+    name = 'type-ignore-comment'
+    priority = -1
+    msgs = {
+        'C0045': (
+            'Please try to Rule out the possibility of \'type: ignore\'.'
+            ' If \'type: ignore\' is really necessary, then add a proper'
+            ' comment.',
+            'mypy-ignore-used',
+            'MyPy ignores should be used with proper comments following'
+            ' the established patterns in the codebase. Except for'
+            ' no-untyped-call MyPy ignore.'
+        ),
+        'C0046': (
+            'Extra comment is present for MyPy type: ignore. Please'
+            ' remove it.',
+            'redundant-type-comment',
+            'No corresponding \'type: ignore\' is found for the type'
+            'ignore comment.'
+        )
+    }
+
+    def visit_module(self, node):
+        """Visit a module to ensure that there is a comment for each MyPy
+        type ignore.
+
+        Args:
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        module_excluded_from_comment_checks = False
+        for directory in self.EXCLUDED_DIRS_HAVING_IGNORE_TYPE_COMMENTS:
+            if re.search(directory, node.root().file):
+                module_excluded_from_comment_checks = True
+        if not module_excluded_from_comment_checks:
+            tokens = pylint_utils.tokenize_module(node)
+            self._process_module_tokens(tokens, node)
+
+    def _process_module_tokens(self, tokens, node):
+        """Custom pylint checker which allows only those MyPy type ignores
+        that are properly documented by a comment in the code. The comment
+        must be close to the actually ignore.
+
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        type_ignore_comment_regex = r'Here we use MyPy ignore because'
+        type_ignore_comment_present = False
+        comment_line_number = 0
+
+        for (token_type, _, (line_num, _), _, line) in tokens:
```

Comment:
What if you had the following all close together in the code:

1. A documentation comment justifying a `type: ignore` comment
2. A documentation comment justifying a `type: ignore` comment
3. A `type: ignore` comment

We should raise an error for the first documentation comment, but I don't think this code will

Reply:
To clarify, here's a code example:

```python
# Here we use MyPy ignore because ...
a = 1
# Here we use MyPy ignore because ...
b = 1  # type: ignore
```

The linter should complain that the first comment has no accompanying `type: ignore` comment, but I don't think the current code will. With the current code, here's what I think will happen:

* We will analyze the first comment, see that it matches `type_ignore_comment_regex`, and set `type_ignore_comment_present = True`.
* We will analyze the second comment, see that it matches `type_ignore_comment_regex`, and set `type_ignore_comment_present = True` even though it already had that value.
* We will analyze the third comment, see that it is a `type: ignore comment` within `ALLOWED_LINES_OF_GAP_IN_COMMENT` lines of the second comment, and so set `type_ignore_comment_present = False` without triggering a lint failure.

Reply:
Now linters are supporting this case as well. but thanks to point that out !!

Reply:
https://github.com/oppia/oppia/actions/runs/3039355891/jobs/4894192430#step:8:755 

PTAL!